src/types.rs

```diff
@@ -810,20 +808,80 @@ fn push_four_digits(dest: &mut WriteBuf, val: u16) -> WriteResult {
     dest.push_byte(b'0' + (val % 10) as u8)
 }
 
+/// A structure representing a date and time. Wrapped by
+/// `UtcTime` and `GeneralizedTime`.
+#[derive(Debug, Clone, PartialEq, Hash, Eq)]
+pub struct DateTime {
+    year: u16,
+    month: u8,
+    day: u8,
+    hour: u8,
+    minute: u8,
+    second: u8,
+}
+
+impl DateTime {
+    pub fn new(
+        year: u16,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: u8,
+    ) -> ParseResult<DateTime> {
+        validate_date(year, month, day)?;
```

Comment:
This is the only call site for `validate_date` now so we could choose to inline it in a follow-up PR.